src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/InstanceRepository.cs

```diff
@@ -96,70 +96,69 @@ public async Task<bool> Delete(Instance item)
             string continuationToken,
             int size)
         {
-            InstanceQueryResponse queryResponse = new InstanceQueryResponse();
-
-            FeedOptions feedOptions = new FeedOptions
+            InstanceQueryResponse queryResponse = new InstanceQueryResponse
             {
-                EnableCrossPartitionQuery = true,
-                MaxItemCount = size,
+                Count = 0,
+                Instances = new List<Instance>()
             };
 
-            if (continuationToken != null)
+            while (queryResponse.Count < size)
             {
-                feedOptions.RequestContinuation = continuationToken;
-            }
-
-            IQueryable<Instance> queryBuilder = _client.CreateDocumentQuery<Instance>(_collectionUri, feedOptions);
+                FeedOptions feedOptions = new FeedOptions
+                {
+                    EnableCrossPartitionQuery = true,
+                    MaxItemCount = size - queryResponse.Count,
+                };
 
-            try
-            {
-                queryBuilder = BuildQueryFromParameters(queryParams, queryBuilder);
-            }
-            catch (Exception e)
-            {
-                queryResponse.Exception = e.Message;
-                return queryResponse;
-            }
+                if (continuationToken != null)
+                {
+                    feedOptions.RequestContinuation = continuationToken;
```

Comment:
Using the same continuationToken every iteration. Not sure if it is wrong or not, but I thought we would need to use the ResponseContinuation value as input in the next iteration.

Reply:
My bad. It is supposed to be reassigned to the value of the ResponseContinuation

Reply:
updated now.